HHS/simpler-grants-gov
# [Issue #2273] create opportunity call to action linking to legacy

## Summary
Fixes #2273

### Time to review: __20 mins__

## Changes proposed

Adds a box on the opportunity detail page with a call to action to visit the legacy site for applying or tracking status on opportunities. See designs in Figma here https://www.figma.com/design/FGxWtAgToKhehLJCiuy1zL/Simpler.Grants.gov-Project?node-id=239-2733&node-type=frame&t=1d7YiGyWrEcOGEuG-0

## Context for reviewers

* In order to fix a bug with testing, I upgraded the testing-library package. There does not seem to have been any change in testing behavior as a result, and all existing tests still pass.

* I created a reusable `OpportunityContentBox` component which could be used in the status widget and the award boxes, but only if we want to have consistent styling on those elements. I've left it unimplemented outside of the CTA for now, but would love to discuss intentions around uniformity and reusability here.

* tightened up some typing

### Test steps

1. visit an opportunity detail page for a posted opportunity
2. _VERIFY_: what you see matches the "posted" screenshot below, paying attention to the verbiage in the "How to apply" box
3. visit an opportunity detail page for a closed, archived or forecasted opportunity
2. _VERIFY_: what you see matches the "forecasted" screenshot below, paying attention to the verbiage in the "How to track application status" box
3. click on links in both cases
4. _VERIFY_ the legacy grants.gov site opens. Note that this should open the corresponding opportunity detail page for a given opportunity, but this won't work with seeded data. We would need to connect to a deployed DB and I'm not sure we're set up for that yet
5. _VERIFY_ styling looks right based on the designs (see link above)

## Additional information

### Posted

<img width="1204" alt="Screenshot 2024-10-22 at 4 18 54 PM" src="https://github.com/user-attachments/assets/e82e4d82-d85f-4fe4-89ae-f3f5e9ee9a13">

### Forecasted


<img width="1260" alt="Screenshot 2024-10-22 at 4 18 38 PM" src="https://github.com/user-attachments/assets/34d213d2-cdb6-40f9-93fd-c1ea7e0b3ade">
